tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -31,44 +24,36 @@ private[scalanative] object LLVM {
    *  @return
    *    The paths of the `.o` files.
    */
-  def compile(config: Config, paths: Seq[Path]): Seq[Path] = {
+  def compile(
+      config: Config,
+      paths: Seq[Path]
+  ): Seq[CompilationResult] = {
+    import NativeSourcePlugin._
+
+    val plugins = LlSourcePlugin +: config.compilerConfig.nativeSourcePlugins
```

Comment:
IIUC the `.ll` files are part of the core functionality—Scala Native will not work without them. So it seems that it should not be optional.